dashboard/test/controllers/levels_controller_test.rb

```diff
@@ -1320,17 +1320,21 @@ def create_level_with_serialized_maze
     assert_response :success
     level.reload
     assert_equal [skill], level.skills
+    assert level.skill_keys.include?(skill.key)
   end
 
   test "remove_skill removes a skill" do
     level = create :level
     skill = create :skill
     create :levels_skill, level: level, skill: skill
     assert_equal [skill], level.skills
+    level.add_skill_key(skill.key)
+    assert level.skill_keys.include?(skill.key)
```

Comment:
I would recommend testing the before state is exactly what you expect (formatted correctly, no extra items) otherwise a bug in add_skill_key could make it look like removing is what's broken.